cadence/grpc_errors.py

```diff
@@ -3,44 +3,51 @@
-            error = error_pb2.WorkflowExecutionAlreadyStartedError()
-            detail.Unpack(error)
-            return WorkflowExecutionAlreadyStartedError(startRequestId=error.start_request_id, runId=error.run_id)
-        else:
-            raise RuntimeError('Unexpected failure: %s' % detail)
\ No newline at end of file
+
+    if status is not None:
+        for detail in status.details:
+            if detail.Is(error_pb2.CancellationAlreadyRequestedError.DESCRIPTOR):
+                return CancellationAlreadyRequestedError()
+            elif detail.Is(error_pb2.ClientVersionNotSupportedError.DESCRIPTOR):
+                error = error_pb2.ClientVersionNotSupportedError()
+                detail.Unpack(error)
+                return ClientVersionNotSupportedError(feature_version=error.feature_version, client_impl=error.client_impl, supported_versions=error.supported_versions)
+            elif detail.Is(error_pb2.DomainAlreadyExistsError.DESCRIPTOR):
+                return DomainAlreadyExistsError()
+            elif detail.Is(error_pb2.DomainNotActiveError.DESCRIPTOR):
+                error = error_pb2.DomainNotActiveError()
+                detail.Unpack(error)
+                return DomainNotActiveError(domainName=error.domain, currentCluster=error.current_cluster, activeCluster=error.active_cluster)
+            elif detail.Is(error_pb2.EntityNotExistsError.DESCRIPTOR):
+                error = error_pb2.EntityNotExistsError()
+                detail.Unpack(error)
+                return EntityNotExistsError(currentCluster=error.current_cluster, activeCluster=error.active_cluster)
+            elif detail.Is(error_pb2.FeatureNotEnabledError.DESCRIPTOR):
+                error = error_pb2.FeatureNotEnabledError()
+                detail.Unpack(error)
+                return FeatureNotEnabledError(featureFlag=error.feature_flag)
+            elif detail.Is(error_pb2.LimitExceededError.DESCRIPTOR):
+                return LimitExceededError()
+            elif detail.Is(error_pb2.QueryFailedError.DESCRIPTOR):
+                return QueryFailedError()
+            elif detail.Is(error_pb2.ServiceBusyError.DESCRIPTOR):
+                return ServiceBusyError(message=status.message)
+            elif detail.Is(error_pb2.WorkflowExecutionAlreadyCompletedError.DESCRIPTOR):
+                return WorkflowExecutionAlreadyCompletedError()
+            elif detail.Is(error_pb2.WorkflowExecutionAlreadyStartedError.DESCRIPTOR):
+                error = error_pb2.WorkflowExecutionAlreadyStartedError()
+                detail.Unpack(error)
+                return WorkflowExecutionAlreadyStartedError(startRequestId=error.start_request_id, runId=error.run_id)
+            else:
+                raise RuntimeError('Unexpected failure: %s' % detail)
+    else:
+        details = e.details()
+        code = e.code()
+        if code == grpc.StatusCode.INVALID_ARGUMENT:
```

Comment:
what happens to the else branch?

Reply:
Good call. I'll return runtime error in a followup PR :)